libs/coin-framework/src/bot/specs.ts

```diff
@@ -118,13 +115,19 @@ export function deviceActionFlow<T extends TransactionCommon>(
             currentStep.stepValueTransform || stepValueTransformDefault;
 
           if (!ignoreAssertionFailure && !disableStrictStepValueValidation) {
-            botTest("deviceAction confirm step '" + stepTitle + "'", () =>
+            botTest("deviceAction confirm step '" + stepTitle + "'", () => {
               expect({
                 [stepTitle]: stepValueTransform(stepValue),
               }).toMatchObject({
-                [stepTitle]: expectedValue(arg, acc).trim(),
-              })
-            );
+                [stepTitle]: expectedValue(arg, acc)
+                  .replace(
+                    "S",
+                    arg.appCandidate.model === DeviceModelId.nanoS ? "S" : ""
+                  )
+                  .trim(),
+                // FIXME: OCR of speculos couldn't retrieve S properly
```

Comment:
Rather put the comment above the code I think? + add the link to the speculos GitHub issue, as discussed :+1: